sys/include/json.h

```diff
@@ -0,0 +1,725 @@
+/*
+ * Copyright (C) 2014-2016  René Kijewski  <rene.kijewski@fu-berlin.de>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+/**
+ * @defgroup    sys_json JSON Data Interchange Format
+ * @ingroup     sys
+ * @brief       A library to read and write JSON
+ *              ([ECMA-404](http://www.ecma-international.org/publications/standards/Ecma-404.htm))
+ *              serialized data.
+ * @{
+ *
+ * @file
+ * @brief       Headers for the JSON module
+ *
+ * @author      René Kijewski <rene.kijewski@fu-berlin.de>
+ */
+
+#ifndef RIOT_SYS_JSON_H_
+#define RIOT_SYS_JSON_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <sys/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Result of JSON read and write functions.
+ */
+typedef enum {
+    JSON_OKAY,                /**< success / do continue */
+    JSON_INVALID_DATA,        /**< invalid marker or type */
+    JSON_PREMATURELY_ENDED,   /**< the stream (abruptly) ended */
+    JSON_READ_UNICODE,        /**< you have to invoke json_read_unicode() */
+} json_result_t;
+
+/* ***************************************************************************
+ * WRITE FUNCTIONS
+ *************************************************************************** */
+
+/**
+ * @defgroup    sys_json_write JSON serialization
+ * @brief       The functions used to serialize data into a JSON stream.
+ * @{
+ */
+
+struct json_write_cookie;
+
+/**
+ * @brief See @ref json_write_cookie.
+ */
+typedef struct json_write_cookie json_write_cookie_t;
+
+/**
+ * @brief Prototype for a JSON write callback.
+ * @param[in] cookie The cookie that was originally provided to json_write_init().
+ * @param[in] buf Buffer to write.
+ * @param[in] len Length of the buffer to write.
+ * @returns `> 0` number of written bytes.
+ * @returns `< 0` to indicate an error.
```

Comment:
Here and in other places of the doc: please make this wall of text a little bit more readable (new lines between "sections", tabular alignment of command and content, ...)